src/macros.rs

```diff
@@ -8,6 +8,66 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __internal_log {
+    (@helper $target:expr, $lvl:expr, $kvs:expr, $($arg:tt)+) => ({
+        let lvl = $lvl;
+        if lvl <= $crate::STATIC_MAX_LEVEL && lvl <= $crate::max_level() {
+            match (::std::format_args!($($arg)+), $target, $kvs) {
+                (args, target, kvs) => if let ::std::option::Option::Some(literal) = ::std::fmt::Arguments::as_str(&args) {
```

Comment:
I don't think this is necessary, since it is turned into an `fmt::Argument<'_>` again when constructing `Record<'_>`.

I think this actually prevents the `Log` implementation to take advantage of `Argument::as_str`.

Reply:
The idea is that passing an `Arguments` value is more costly than passing an `str` value: <https://godbolt.org/z/ehPcoqsov>. I expect an application may contain a lot of logs that only contains string literals, so If we can reduce the size of each log, the total binary size could been reduced. Essentially, this implementation tries to bring back an old optimization that is reverted by #446.

Reply:
You can pass a reference to Arguments, then dereference it when building Records.
It implements Copy, so you can avoid copying it in each function while preserving the ability to take advantages of Arguments::as_str

Reply:
Aside from copying, building an `Argument` object also has costs: <https://godbolt.org/z/a1eqr4bq5>.

Reply:
It seems that passing `Arguments` and padding `&Arguments` have the same cost: <https://godbolt.org/z/Wz35xYrsr>.